BoxContentSDK/BoxContentSDK/Models/BOXRepresentation.m

```diff
@@ -41,27 +46,26 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
                                                                     nullAllowed:NO];
         self.details = detailsJSON;
         
-        NSDictionary *linksJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyLinks
+        NSDictionary *contentJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyContent
                                                                  inDictionary:JSONResponse
                                                               hasExpectedType:[NSDictionary class]
                                                                   nullAllowed:NO];
         
-        NSDictionary *contentJSON = linksJSON[BOXAPIObjectKeyContent];
-        
-        NSString *contentURLString = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyURL
+        NSString *contentURLString = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyURLTemplate
                                                              inDictionary:contentJSON
                                                           hasExpectedType:[NSString class]
                                                               nullAllowed:NO];
         if (contentURLString.length > 0) {
+            if ([self.type isEqualToString:BOXRepresentationTypeHLS]) {
```

Comment:
This should probably return `nil` (which I think is the error indicator) if it's not HLS, because otherwise, your `[NSURL URLWithString:contentURLString]` will fail below due to invalid url characters (`{}` are guaranteed to be part of the url_template).

I know this isn't changing, but `name` or `representationName` would be a better identifier than `type`.